src/devices/Bmp280/Bmp280.cs

```diff
@@ -0,0 +1,562 @@
+            _calibrationData.DigP8 = (short)Read16((byte)Registers.REGISTER_DIG_P8);
+            _calibrationData.DigP9 = (short)Read16((byte)Registers.REGISTER_DIG_P9);
+
+        }
+
+        /// <summary>
+        /// Set mode sleep (no measurements)
+        /// </summary>
+        public void SetModeSleep()
+        {
+            byte status = Read8((byte)Registers.REGISTER_CONTROL);
+            //clear last two bits
+            status = (byte)(status & 0b1111_1100);
+            _i2cDevice.Write(new[] { (byte)Registers.REGISTER_CONTROL, status });
+        }
+
+        /// <summary>
+        /// Sets mode to forced (device goes to sleep after one reading)
+        /// </summary>
+        public void SetModeForced()
+        {
+            byte status = Read8((byte)Registers.REGISTER_CONTROL);
+            //clear last two bits
+            status = (byte)(status & 0b1111_1100);
+            //set one of the last two bits
+            status = (byte)(status | 0b0000_0010);
+            _i2cDevice.Write(new[] { (byte)Registers.REGISTER_CONTROL, status });
+        }
+
+        /// <summary>
+        /// Sets mode to normal (continuous measurements)
+        /// </summary>
+        public void SetModeNormal()
+        {
+            byte status = Read8((byte)Registers.REGISTER_CONTROL);
+            //set last two bits to 11
+            status = (byte)(status | 0b0000_0011);
+            _i2cDevice.Write(new[] { (byte)Registers.REGISTER_CONTROL, status });
+        }
+
+        public PowerMode ReadPowerMode()
+        {
+            byte status = Read8((byte)Registers.REGISTER_CONTROL);
+            status = (byte)(status & 0b000_00011);
+            if (status == (byte)PowerMode.Normal)
+            {
+                return PowerMode.Normal;
+            }
+            else if (status == (byte)PowerMode.Sleep)
+            {
+                return PowerMode.Sleep;
+            }
+            else
+            {
+                return PowerMode.Forced;
+            }
+        }
+
+        /// <summary>
+        /// Sets the temperature sampling to the given value
+        /// </summary>
+        /// <param name="sampling"></param>
+        public void SetTemperatureSampling(Sampling sampling)
+        {
+            byte status = Read8((byte)Registers.REGISTER_CONTROL);
+            status = (byte)(status & 0b0001_1111);
+            status = (byte)(status | (byte)sampling << 5);
+            _i2cDevice.Write(new[] { (byte)Registers.REGISTER_CONTROL, status });
+        }
+
+        /// <summary>
+        /// Get the sample rate for temperature measurements
+        /// </summary>
+        /// <returns></returns>
+        public Sampling ReadTemperatureSampling()
+        {
+            byte status = Read8((byte)Registers.REGISTER_CONTROL);
+            status = (byte)((status & 0b1110_0000) >> 5);
+            return ByteToSampling(status);
+        }
+
+        private Sampling ByteToSampling(byte value)
+        {
```

Comment:
Given that you are controlling all of the callsites for this method, should you simplify this to something like:
```c#
private Sampling ByteToSampling(byte value) => (Sampling)value;
```

Reply:
According to the datasheet values >5 are ultra high resolution so just casting to Sampling won't return the correct Sampling value for all cases.

Is something like this preferred?

```C#
private Sampling ByteToSampling(byte value)
{
    //Values >=5 equals UltraHighResolution
    if (value >= 5)
    {
        return Sampling.UltraHighResolution;
     }
     return (Sampling) value;            
}
```

Reply:
sure, that sounds good to me since, as I said earlier, you control all the call sites so you know that you won't get some weird input passed in.

Reply:
Just want to be sure to return the correct sampling value is returned if a higher value is read from the device.